galsim/correlatednoise.py

```diff
@@ -117,6 +121,106 @@ def __init__(self, rng, gsobject, wcs=PixelScale(1.0)):
         # noise field can get convolved with other GSObjects making is_analytic_x False
         self._variance_cached = None
 
+    @classmethod
+    def from_file(cls, file_name, pixel_scale, rng=None, variance=0., x_interpolant=None,
+                  gsparams=None):
+        """Read a correlated noise profile from a file.
+
+        The file should contain an image of the correlation.
+
+            * The image should be square with odd size in each direction.
+            * The central pixel corresponds to the zero-lag correlation, i.e. the variance.
+              Call this pixel element (0,0).
+            * The other pixels (i,j) give the cross-correlation of the noise for pixels separated
+              by i pixels in the x direction (columns) and j pixels in the y direction (rows).
+            * The image therefore must be 180 degree rotationally symmetric.  i.e. the value at
+              (i,j) must be the same as at (-i,-j).
+
+        The pixel_scale is also required and defines the pixel scale of the original image.
+
+        The default ``x_interpolant`` is a ``galsim.Linear()``, which uses bilinear interpolation.
+        The use of this interpolant is an approximation that gives good empirical results without
+        requiring internal convolution of the correlation function profile by a `Pixel` object when
+        applying correlated noise to images: such an internal convolution has been found to be
+        computationally costly in practice, requiring the Fourier transform of very large arrays.
+
+        The use of the bilinear interpolants means that the representation of correlated noise will be
+        noticeably inaccurate in at least the following two regimes:
+
+        1. If the pixel scale of the desired final output (e.g. the target image of
+           `BaseCorrelatedNoise.drawImage`, `BaseCorrelatedNoise.applyTo` or
+           `BaseCorrelatedNoise.whitenImage`) is small relative to the separation between pixels in
+           the ``image`` used to instantiate ``cn`` as shown above.
+        2. If the `BaseCorrelatedNoise` instance ``cn`` was instantiated with an image of scale
+           comparable to that in the final output, and ``cn`` has been rotated or otherwise transformed
+           (e.g.  via the `BaseCorrelatedNoise.rotate`, `BaseCorrelatedNoise.shear` methods; see below).
+
+        Conversely, the approximation will work best in the case where the correlated noise used to
+        instantiate the ``cn`` is taken from an input image for which ``image.scale`` is smaller than
+        that in the desired output.  This is the most common use case in the practical treatment of
+        correlated noise when simulating galaxies from space as observed in ground-based surveys.
+
+        Changing from the default bilinear interpolant is made possible, but not recommended.  For more
```

Comment:
This paragraph makes it sound like the PR discusses interpolant choices, which I was curious about, so I went there.  I couldn't find anything on it there - did I miss it?  I was actually wondering how much testing we've done with modifying the interpolants for these CorrelatedNoise objects.

Reply:
Well, I tracked it down.  Here is the relevant thread:  https://github.com/GalSim-developers/GalSim/pull/452/files#r5701561.  I'm not sure why it shows up in files, but not conversation...

However, this probably isn't super useful to point users to go looking at.  It's basically me and Barney repeatedly saying we don't really understand what the right thing is here, and that Linear just seems to work out the best.  Pretty unsatisfying, really.  So I'm just going to change this to something more generic: "Changing from the default bilinear interpolant is made possible, but not recommended. In our validation tests, we found that the `Linear` interpolant usually gave the most accurate results."

(BTW, this text is all copied verbatim from the `getCOSMOSNoise` function.)